playbooks/06-deploy-va.yml

```diff
@@ -0,0 +1,124 @@
+        tasks_from: load_env_definition.yml
+
+    - name: Fetch controller-0 network facts
+      when: "'ansible_interfaces' not in hostvars['controller-0']"
+      ansible.builtin.setup:
+        gather_subset: network
+      delegate_to: controller-0
+      delegate_facts: true
+
+    - name: Create nova migration keypair
+      register: _nova_key
+      community.crypto.openssh_keypair:
+        comment: "nova migration"
+        path: >-
+          {{ (cifmw_basedir,
+             'artifacts/nova_migration_key') | path_join
+          }}
+        type: "ecdsa"
+
+    - name: Generate needed facts out of local files
+      vars:
+        _ifaces_vars: >-
+          {{
+            hostvars['controller-0'].ansible_interfaces |
+            map('regex_replace', '^(.*)$', 'ansible_\1')
+          }}
+        _controller_host: "{{ hostvars['controller-0'].ansible_host }}"
+        _ipv4_network_data: >-
+          {{
+            hostvars['controller-0'] | dict2items |
+            selectattr('key', 'in', _ifaces_vars) |
+            selectattr('value.ipv4.address', 'defined') |
+            selectattr('value.ipv4.address', 'equalto', _controller_host) |
+            map(attribute='value.ipv4') | first | default({})
+          }}
+      ansible.builtin.set_fact:
+        cifmw_ci_gen_kustomize_values_ssh_authorizedkeys: >-
+          {{ lookup('file', '~/.ssh/authorized_keys', rstrip=False) }}
+        cifmw_ci_gen_kustomize_values_ssh_private_key: >-
+          {{ lookup('file', '~/.ssh/id_cifw', rstrip=False) }}
+        cifmw_ci_gen_kustomize_values_ssh_public_key: >-
+          {{ lookup('file', '~/.ssh/id_cifw.pub', rstrip=False) }}
+        cifmw_ci_gen_kustomize_values_migration_pub_key: >-
+          {{ lookup('file', _nova_key.filename ~ '.pub', rstrip=False)}}
+        cifmw_ci_gen_kustomize_values_migration_priv_key: >-
+          {{ lookup('file', _nova_key.filename, rstrip=False) }}
+        cifmw_ci_gen_kustomize_values_sshd_ranges: >-
+          {{
+            [cifmw_networking_env_definition.networks.ctlplane.network_v4] +
+            (
+              [
+                 _ipv4_network_data.network + '/' + _ipv4_network_data.prefix
+              ]
+            ) if (_ipv4_network_data | length > 0) else []
+          }}
+
+    - name: Load architecture automation file
+      register: _automation
+      ansible.builtin.slurp:
+        path: "{{ cifmw_architecture_automation_file }}"
+
+    - name: Prepare automation data
+      vars:
+        _parsed: "{{ _automation.content | b64decode | from_yaml }}"
+      ansible.builtin.set_fact:
+        cifmw_deploy_va_steps: >-
+          {{ _parsed['vas'][cifmw_architecture_va_scenario] }}
+
+    - name: Check requirements
+      ansible.builtin.import_role:
+        name: kustomize_deploy
+        tasks_from: check_requirements.yml
+
+    - name: Deploy OSP operators
+      ansible.builtin.import_role:
+        name: kustomize_deploy
+        tasks_from: install_operators.yml
```

Comment:
The "kustomization" capabilities we have here are really poor, basically none. How is a user going to kustomize this? https://github.com/openstack-k8s-operators/architecture/blob/main/lib/olm-openstack/kustomization.yaml (needs work in the architecture repo too, so the scope of this comment is not only this PR)

Reply:
+1

Reply:
I already talked about this with @rebtoor - there's a need to extend a bit the `install_operators.yml` to allow kustomization. This won't be part of the current patch though, the feature needs to be implemented first.